src/libraries/System.Linq/src/System/Linq/Single.cs

```diff
@@ -69,7 +69,7 @@ public static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source,
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
             }
 
-            if (source is IList<TSource> list)
+            if (source is IReadOnlyList<TSource> list)
```

Comment:
There were a few new LINQ operators added since these changes were initially made. I think the answer is "no", but are there any additional uses of ICollection/IList/etc. that got added in the interim that should also be updated (since they wouldn't naturally happen as part of a revert)?

Reply:
> follow up PR

Sure

> we could add IReadOnlyList<T> optimized paths too (can't share them due to no CopyTo)

Meh. I don't think it's worthwhile. It's one thing to expand places we currently check for IList to "for free" get any read-only lists that don't also implement IList (most do). It's another to add additional code and cost to check for that rare situation.